src/Foundatio.Parsers.ElasticQueries/Visitors/CombineAggregationsVisitor.cs

```diff
@@ -26,71 +25,7 @@ public override async Task VisitAsync(GroupNode node, IQueryVisitorContext conte
         {
             var aggregation = await child.GetAggregationAsync(() => child.GetDefaultAggregationAsync(context));
             if (aggregation == null)
-            {
-                var termNode = child as TermNode;
-                if (termNode != null && termsAggregation != null)
-                {
-                    // TODO: Move these to the default aggs method using a visitor to walk down the tree to gather them but not going into any sub groups
-                    if (termNode.Field == "@exclude")
-                    {
-                        termsAggregation.Exclude = termsAggregation.Exclude.AddValue(termNode.UnescapedTerm);
-                    }
-                    else if (termNode.Field == "@include")
-                    {
-                        termsAggregation.Include = termsAggregation.Include.AddValue(termNode.UnescapedTerm);
```

Comment:
How come we wouldn't have been able to just update this to handle the regex expressions?

Reply:
@ejsmith Regex expressions in include/exclude were completely not working. 

But there is another reason why we need to move that population of aggregations into GetDefaultAggregationAsync. We have implemented handling of nested fields in aggregations by adding custom ChainableQueryVisitor. And in order to support @include/@exclude/@missing and etc for aggs with nested fields we had to copy that population logic from CombineAggregationVisitor to our custom visitor OR move that logic to GetDefaultAggregationAsync and get everything we need for free.